test/tstRay.cpp

```diff
@@ -225,3 +225,32 @@ BOOST_AUTO_TEST_CASE(overlap_distance_sphere,
   BOOST_TEST(overlapDistance(Ray{{half_sqrtf_3, 0.5, 0}, {1, 0, 0}},
                              unit_sphere) == 0.f);
 }
+
+#define STATIC_ASSERT(cond) static_assert(cond, "");
+BOOST_AUTO_TEST_CASE(make_euclidean_vector)
+{
+  using ArborX::Experimental::makeVector;
+  using ArborX::Experimental::Vector;
+  STATIC_ASSERT((makeVector({0, 0, 0}, {1, 2, 3}) == Vector{1, 2, 3}));
+  STATIC_ASSERT((makeVector({1, 2, 3}, {4, 5, 6}) == Vector{3, 3, 3}));
+}
+
+BOOST_AUTO_TEST_CASE(dot_product)
+{
+  using ArborX::Experimental::dotProduct;
+  STATIC_ASSERT(dotProduct({1, 0, 0}, {1, 0, 0}) == 1);
+  STATIC_ASSERT(dotProduct({1, 0, 0}, {0, 1, 0}) == 0);
+  STATIC_ASSERT(dotProduct({1, 0, 0}, {0, 0, 1}) == 0);
+  STATIC_ASSERT(dotProduct({1, 1, 1}, {1, 1, 1}) == 3);
+}
+
+BOOST_AUTO_TEST_CASE(cross_product)
+{
+  using ArborX::Experimental::crossProduct;
+  using ArborX::Experimental::Vector;
+  STATIC_ASSERT((crossProduct({1, 0, 0}, {1, 0, 0}) == Vector{0, 0, 0}));
```

Comment:
What's the point of having this check when you also have `crossProduct(a, a)` with `a = {1, 1, 1}` later?

Reply:
doesn't hurt either